src/DynamicRupture/FrictionLaws/RateAndState.h

```diff
@@ -408,10 +382,10 @@ class RateAndStateBase : public BaseFrictionLaw<RateAndStateBase<Derived>> {
         // for compiling reasons ASINH(X)=LOG(X+SQRT(X^2+1))
 
         // calculate friction coefficient
-        tmp2[pointIndex] = tmp[pointIndex] * slipRateTest[pointIndex];
-        muF[pointIndex] = a[ltsFace][pointIndex] * misc::asinh(tmp2[pointIndex]);
-        dMuF[pointIndex] =
-            a[ltsFace][pointIndex] / sqrt(1.0 + misc::power<2>(tmp2[pointIndex])) * tmp[pointIndex];
+        muF[pointIndex] = static_cast<Derived*>(this)->updateMu(
```

Comment:
Don't you think this implementation could be slower because 
```      
tmp[pointIndex] = 0.5 / this->drParameters.rsSr0 *
                        exp(localStateVariable[pointIndex] / a[ltsFace][pointIndex]);
```
is now computed twice (for mu and muf), and this is a part very critical of the RS code, being repeated many times?

Reply:
I wanted to provide an interface as general as possible here. Once we have all functionality working, we'll come back to performance.